plugins/plugin.php

```diff
@@ -32,42 +39,55 @@ function __construct($plugins) {
 		$this->plugins = $plugins;
 		//! it is possible to use ReflectionObject to find out which plugins defines which methods at once
 	}
-	
+
+	/** Register additional plugin after instantiating the class
+	 * @param $plugin
+	 */
+	function addPlugin($plugin){
+		$this->plugins[] = $plugin;
+	}
+
 	function _callParent($function, $args) {
 		return call_user_func_array(array('parent', $function), $args);
 	}
 	
 	function _applyPlugin($function, $args) {
-		foreach ($this->plugins as $plugin) {
-			if (method_exists($plugin, $function)) {
-				switch (count($args)) { // call_user_func_array() doesn't work well with references
-					case 0: $return = $plugin->$function(); break;
-					case 1: $return = $plugin->$function($args[0]); break;
-					case 2: $return = $plugin->$function($args[0], $args[1]); break;
-					case 3: $return = $plugin->$function($args[0], $args[1], $args[2]); break;
-					case 4: $return = $plugin->$function($args[0], $args[1], $args[2], $args[3]); break;
-					case 5: $return = $plugin->$function($args[0], $args[1], $args[2], $args[3], $args[4]); break;
-					case 6: $return = $plugin->$function($args[0], $args[1], $args[2], $args[3], $args[4], $args[5]); break;
-					default: trigger_error('Too many parameters.', E_USER_WARNING);
-				}
-				if ($return !== null) {
-					return $return;
+		if($this->propagation) {
```

Comment:
This can be moved to the AdminerBootstrapLike like:

```php
function _applyPlugin($function, $args) {
    if($this->propagation) {
       parent::_applyPlugin($function, $args);
    }
   ...
}
```

You know.

Sorry for the delay, I did not know what to do with it...

Reply:
Reset the AdminerPlugin according with the current version...